cmd/juju/commands/upgradecontroller.go

```diff
@@ -215,17 +216,16 @@ func (c *upgradeControllerCommand) upgradeCAASController(ctx *cmd.Context) error
 // agent and client versions, and the list of currently available operator images, will
 // always be accurate; the chosen version, and the flag indicating development
 // mode, may remain blank until uploadTools or validate is called.
-func (c *upgradeControllerCommand) initCAASVersions(
-	controllerCfg controller.Config, agentVersion version.Number, filterOnPrior bool,
+func initCAASVersions(
+	controllerCfg controller.Config, desiredVersion, agentVersion version.Number, filterOnPrior bool,
 ) (*upgradeContext, error) {
-	if c.Version == agentVersion {
+	if desiredVersion == agentVersion {
 		return nil, errUpToDate
 	}
-	c.upgradeMessage = "upgrade to this version by running\n    juju upgrade-controller"
 
 	filterVersion := jujuversion.Current
-	if c.Version != version.Zero {
-		filterVersion = c.Version
+	if desiredVersion != version.Zero {
```

Comment:
do we need raise errors if it's a downgrade?

Reply:
This filter logic was copied across from the upgrade-juju command. Not sure exactly the reason for it but didn't want to mess with it as the upgrade stuff is complex and can easily break.